cdap-common/src/main/java/co/cask/cdap/internal/test/AppJarHelper.java

```diff
@@ -47,23 +47,31 @@ private AppJarHelper() {
 
   public static Location createDeploymentJar(LocationFactory locationFactory, Class<?> clz, Manifest manifest,
                                              File... bundleEmbeddedJars) throws IOException {
+    return createDeploymentJar(locationFactory, clz, manifest, new ClassAcceptor() {
+      final Set<String> visibleResources = ProgramResources.getVisibleResources();
 
-    final Set<String> visibleResources = ProgramResources.getVisibleResources();
-    // Exclude all classes that are visible form the system to the program classloader.
-    ApplicationBundler bundler = new ApplicationBundler(new ClassAcceptor() {
       @Override
       public boolean accept(String className, URL classUrl, URL classPathUrl) {
         if (visibleResources.contains(className.replace('.', '/') + ".class")) {
           return false;
         }
-        // Exclude all Spark classes in the app jar.
-        // We keep the Scala classes in case the app is not Spark but uses Scala in unit-test (CDAP-5168)
+        if (className.startsWith("scala.")) {
```

Comment:
why is this needed? scala classes would have been accepted without this check right?